bpf/process/bpf_generic_lsm_ima_bprm.c

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/* Copyright Authors of Cilium */
+
+#include "vmlinux.h"
+#include "api.h"
+
+#define GENERIC_LSM
+
+#include "compiler.h"
+#include "bpf_event.h"
+#include "bpf_task.h"
+#include "bpf_lsm_ima.h"
+#include "retprobe_map.h"
+#include "types/basic.h"
+
+char _license[] __attribute__((section("license"), used)) = "Dual BSD/GPL";
+
+struct filter_map_value {
+	unsigned char buf[FILTER_SIZE];
+};
+
+struct {
+	__uint(type, BPF_MAP_TYPE_ARRAY);
+	__uint(max_entries, 1);
+	__type(key, int);
+	__type(value, struct filter_map_value);
+} filter_map SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_ARRAY);
+	__uint(max_entries, 1);
+	__type(key, __u32);
+	__type(value, struct event_config);
+} config_map SEC(".maps");
+
+__attribute__((section("lsm.s/generic_lsm_ima_bprm"), used)) int
+BPF_PROG(ima_bprm, struct linux_binprm *bprm)
+{
+	struct ima_hash hash;
+	__u64 pid_tgid = get_current_pid_tgid();
+	struct ima_hash *dummy = map_lookup_elem(&ima_hash_map, &pid_tgid);
+
+	if (dummy && dummy->state == 1) {
+#ifdef __V61_BPF_PROG
+		hash.algo = ima_file_hash(bprm->file, &hash.value, 64);
+#else
+		hash.algo = ima_inode_hash(bprm->file->f_inode, &hash.value, 64);
+#endif
+		hash.state = 3; // 1 + 2
```

Comment:
curious, any reason why it's + 2 ?

Reply:
I waited for this question:)). It seems, that I've done a bit overengineering. My idea was that if we sum different values, we can understand which program was not triggered (but we can't know if the order of execution was correct). I think just incrementing in is more intuitive. Anyway if error happens, we will need to debug. I'll put plus one at each stage.